cpp/src/fil/infer.cu

```diff
@@ -308,18 +319,18 @@ struct tree_aggregator_t<NITEMS, CATEGORICAL_LEAF> {
   int* votes;
   int num_classes;
 
-  static size_t smem_finalize_footprint(int num_classes) {
-    return sizeof(int) * num_classes * NITEMS;
+  static size_t smem_finalize_footprint(shmem_size_params params) {
+    return smem_accumulate_footprint(params.num_classes) +
+           cols_shmem_size(params);
```

Comment:
this fixes a latent bug in FIL, hence the need to replace `num_classes` with `shmem_size_params`